src/Stryker.Core/Stryker.Core/DashboardCompare/GitInfoProvider.cs

```diff
@@ -5,115 +5,123 @@
-            else
-            {
-                Repository = CreateRepository();
-            }
+            this.Repository = repository ?? this.CreateRepository();
         }
 
         public string GetCurrentBranchName()
         {
-            if (Repository?.Branches != null)
-            {
-                foreach (var branch in Repository.Branches)
-                {
-                    if (branch.IsCurrentRepositoryHead)
-                    {
-                        return branch.FriendlyName;
-                    }
-                }
-            }
-
-            return string.Empty;
+            return this.Repository?.Branches?.FirstOrDefault(x => x.IsCurrentRepositoryHead)?.FriendlyName ?? string.Empty;
         }
 
         private IRepository CreateRepository()
         {
-            if (string.IsNullOrEmpty(RepositoryPath))
+            if (string.IsNullOrEmpty(this.RepositoryPath))
             {
                 throw new StrykerInputException("Could not locate git repository. Unable to determine git diff to filter mutants. Did you run inside a git repo? If not please disable the --diff feature.");
             }
-            else
-            {
-                return new Repository(RepositoryPath);
-            }
+
+            return new Repository(this.RepositoryPath);
         }
 
-        private void Checkout()
+        private void Checkout(GitSourceKinds gitSourceKind)
         {
             try
             {
-                var branch = Repository.CreateBranch(_options.GitSource, $"origin/{_options.GitSource}");
-
-                Commands.Checkout(Repository, branch);
+                var branchName = gitSourceKind == GitSourceKinds.FriendlyBranchName ? this._options.GitSource : this.GetFriendlyName(this._options.GitSource);
 
-                var currentBranch = Repository.CreateBranch(_options.ProjectVersion, $"origin/{_options.ProjectVersion}");
+                this._logger.LogDebug($"Creating branch ${branchName} with committish origin/{branchName}");
+                var branch = this.Repository.CreateBranch(this._options.ProjectVersion, $"origin/{this._options.ProjectVersion}");
+                this._logger.LogDebug($"Checking out branch ${branchName}");
+                Commands.Checkout(this.Repository, branch);
 
-                Commands.Checkout(Repository, currentBranch);
+                this._logger.LogDebug($"Creating branch ${this._options.ProjectVersion} with committish origin/{this._options.ProjectVersion}");
+                var currentBranch = this.Repository.CreateBranch(this._options.ProjectVersion, $"origin/{this._options.ProjectVersion}");
+                this._logger.LogDebug($"Checking out branch ${this._options.ProjectVersion}");
+                Commands.Checkout(this.Repository, currentBranch);
             }
-            catch
+            catch (Exception e)
             {
+                this._logger.LogDebug($"Something went wrong during checkout.\n{e.Message}");
```

Comment:
As I understand it, an exception can occur here when the checkout was already performed @Stijn-Rutten

Reply:
Thats true, If I recall correctly there was something about the way libgit threw exceptions during checkout that made it hard to check what error was thrown. Don't know exactly what the problem was. Maybe something like an internal access modifier on the exception.